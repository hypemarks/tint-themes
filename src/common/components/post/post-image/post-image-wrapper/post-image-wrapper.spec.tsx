import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import PostImageWrapper from './post-image-wrapper';

const onClick = jest.fn();

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Test <PostImageWrapper /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(<PostImageWrapper onClick={onClick} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should render anchor tag with specific href', () => {
    const wrapper = render(<PostImageWrapper href={'somesortofstring'} />);
    const postContentLinkAnchor = wrapper.getByTestId('post-image-wrapper-anchor');
    expect(postContentLinkAnchor).toBeInTheDocument();
    expect(postContentLinkAnchor).toHaveAttribute('href', 'somesortofstring');
  });

  it('should render anchor tag with specific href and not fire onClick', () => {
    const wrapper = render(<PostImageWrapper onClick={onClick} />);
    const postContentLinkAnchor = wrapper.getByTestId('post-image-wrapper-anchor');
    fireEvent.click(postContentLinkAnchor);
    expect(onClick).not.toBeCalled();
  });

  it('should render div as button with specific text content', () => {
    const someSortOfText = 'textttt';
    const wrapper = render(
      <PostImageWrapper onClick={onClick} src={'http://www.tintup.com'}>
        {someSortOfText}
      </PostImageWrapper>
    );
    const postImageWrapperButton = wrapper.getByTestId('post-source-icon-image-wrapper-button');
    expect(postImageWrapperButton).toHaveTextContent(someSortOfText);
  });
});
