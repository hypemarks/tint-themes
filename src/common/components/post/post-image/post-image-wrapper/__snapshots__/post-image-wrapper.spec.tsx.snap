// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PostImageWrapper /> Component should render snapshot 1`] = `
<DocumentFragment>
  <a
    class="post-image-wrapper--no-image"
    data-testid="post-image-wrapper-anchor"
    rel="noopener noreferrer"
    target="_blank"
  />
</DocumentFragment>
`;
